aea/configurations/data_types.py

```diff
@@ -498,6 +498,10 @@ def without_hash(
         """Returns a `PublicId` object with same parameters."""
         return PublicId(self.author, self.name, self.version)
 
+    def with_hash(self, package_hash: str) -> "PublicId":
```

Comment:
Hm, I thought we had some methods like this already..

Reply:
We had `without_hash`